src/async_fuse/memfs/kv_engine.rs

```diff
@@ -403,17 +677,46 @@ macro_rules! retry_txn {
 #[async_trait]
 impl KVEngine for EtcdKVEngine {
     #[must_use]
-    fn new(etcd_client: etcd_client::Client) -> Self {
-        EtcdKVEngine {
-            client: etcd_client,
-        }
+    async fn new(end_points: Vec<String>) -> DatenLordResult<Self> {
+        let client = etcd_client::Client::connect(end_points, None).await?;
+        Ok(Self { client })
     }
     async fn new_meta_txn(&self) -> Box<dyn MetaTxn + Send> {
         Box::new(EtcdTxn::new(self.client.clone()))
     }
+
+    async fn lease_grant(&self, ttl: i64) -> DatenLordResult<i64> {
+        let mut client = self.client.clone();
+        let timeout_sec = check_ttl(ttl)
+            .with_context(|| "timeout_sec should be >=1s, please fix the call".to_owned())?;
+        Ok(client
+            .lease_grant(timeout_sec, None)
+            .await
+            .with_context(|| "failed to get lease at `MetaTxn::lock`".to_owned())?
+            .id())
+    }
+
+    async fn range(&self, mut key_range: KeyRange) -> DatenLordResult<Vec<(Vec<u8>, Vec<u8>)>> {
+        let resp = self
+            .client
+            .kv_client()
+            .get(
+                key_range.take_key(),
+                Some(GetOptions::new().with_range(key_range.take_range_end())),
+            )
+            .await?;
+        let kvs = resp.kvs();
+        let mut result = Vec::with_capacity(kvs.len());
+        for kv in kvs {
+            result.push((kv.key().to_vec(), kv.value().to_vec()));
+        }
+        Ok(result)
+    }
+
     /// Distribute lock - lock
     /// - `timeout_sec` should be >=1s
-    async fn lock(&self, key: &LockKeyType, timeout_sec: Duration) -> DatenLordResult<()> {
+    /// - `timeout_sec` should be >=1s
+    async fn lock(&self, key: &LockKeyType, timeout_sec: Duration) -> DatenLordResult<Vec<u8>> {
         let mut client = self.client.clone();
```

Comment:
`clone` client for every operation doesn't make sence. What' the reason?

Reply:
@ActivePeter